elastic/kibana
# [kbn-es] Only enable security for trial

Prior to this PR, you would be locked out of Kibana if you attempted to start a trial when using `kbn-es`. This is due to the fact we were enabling security when running with a basic or trial license. We should only be enabling security for a trial license. The reason security is disabled for trial by default, is to allow for starting a trial at will since security requires additional configuration to set up authentication.

To test:

- Start ES with `yarn es snapshot`
- Start Kibana
- Management > License Management > Start Trial
- Ensure the trial starts and security is not enabled, resulting in being signed out of Kibana.

@elastic/kibana-operations 

Closes: https://github.com/elastic/kibana/issues/20799
